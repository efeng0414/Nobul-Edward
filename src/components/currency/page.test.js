/* eslint-disable no-undef */ // We're using "global"
import React from "react";
import PageCurrencyWrapped from "./page";
import renderer from "react-test-renderer";
import { MemoryRouter } from "react-router";

describe("Page Currency component", () => {
  const { WrappedComponent: PageCurrency } = PageCurrencyWrapped;
  const componentProps = {
    location: {
      search: "?FL"
    },
    value: 4000000.32
  };

  const containerJSX = <PageCurrency {...componentProps} />;
  const snapshotJSX = (
    <MemoryRouter initialEntries={[{ search: "?FL" }]}>
      {containerJSX}
    </MemoryRouter>
  );

  const componentMount = global.Enzyme.mount(containerJSX);

  it("should match existing snapshot", () => {
    const containerSnapshot = renderer.create(snapshotJSX).toJSON();
    expect(containerSnapshot).toMatchSnapshot();
  });

  describe("Checking state correctly loaded", () => {
    it("should store correct state", () => {
      expect(componentMount.state().stateCode).toBe("FL");
    });

    it("should update to correct state", () => {
      componentMount.setProps({
        location: {
          search: "?ON"
        }
      });
      expect(componentMount.state().stateCode).toBe("ON");
    });
  });
});
