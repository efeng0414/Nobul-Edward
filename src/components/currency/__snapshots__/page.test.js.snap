// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page Currency component should match existing snapshot 1`] = `
<span
  className="currency"
>
  <span
    className="currency__identifier"
    dangerouslySetInnerHTML={
      Object {
        "__html": "<span class='currency__code'>US</span>$",
      }
    }
  />
  <span
    className="currency__value"
  >
    4,000,000.32
  </span>
</span>
`;
